From d5a5b28cfb6c04ef102babe1e739d7a9eee62ede Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:13:28 -0400
Subject: [powerpc] Avoid circular dependency with zImage.%

Message-id: <1410545655-205645-580-git-send-email-dzickus@redhat.com>
Patchwork-id: 94583
O-Subject: [RHEL7 PATCH 579/626] powerpc: Avoid circular dependency with zImage.%
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 699c659b49b5c7cf601fe454683841df16495c3e
Author: Michal Marek <mmarek@suse.cz>
Date:   Wed Jun 11 13:53:48 2014 +0200

    powerpc: Avoid circular dependency with zImage.%

    The rule to create the final images uses a zImage.% pattern.
    Unfortunately, this also matches the names of the zImage.*.lds linker
    scripts, which appear as a dependency of the final images. This somehow
    worked when $(srctree) used to be an absolute path, but now the pattern
    matches too much. List only the images from $(image-y) as the target of
    the rule, to avoid the circular dependency.

    Reported-and-tested-by: Mike Qiu <qiudayu@linux.vnet.ibm.com>
    Signed-off-by: Michal Marek <mmarek@suse.cz>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/boot/Makefile b/arch/powerpc/boot/Makefile
index 0ee6da3..616cc98 100644
--- a/arch/powerpc/boot/Makefile
+++ b/arch/powerpc/boot/Makefile
@@ -330,8 +330,8 @@ $(addprefix $(obj)/, $(initrd-y)): $(obj)/ramdisk.image.gz
 $(obj)/zImage.initrd.%: vmlinux $(wrapperbits)
  $(call if_changed,wrap,$*,,,$(obj)/ramdisk.image.gz)
 
-$(obj)/zImage.%: vmlinux $(wrapperbits)
- $(call if_changed,wrap,$*)
+$(addprefix $(obj)/, $(sort $(filter zImage.%, $(image-y)))): vmlinux $(wrapperbits)
+ $(call if_changed,wrap,$(subst $(obj)/zImage.,,$@))
 
 # dtbImage% - a dtbImage is a zImage with an embedded device tree blob
 $(obj)/dtbImage.initrd.%: vmlinux $(wrapperbits) $(obj)/%.dtb
-- 
1.7.1